modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/integration/UserDefinedFunctionsIntegrationTest.java

```diff
@@ -61,6 +67,64 @@ public class UserDefinedFunctionsIntegrationTest extends AbstractBasicIntegratio
+            .setSqlFunctionClasses(OverrideSystemFunctionLibrary.class)
+            .setQueryEntities(F.asList(new QueryEntity(Integer.class, Employer.class).setTableName("emp_own")))
+        );
+
+        // Make sure that the new functions didn't affect schema 'PUBLIC'.
+        assertQuery("SELECT UPPER(?)").withParams("abc").returns("ABC").check();
+        assertQuery("select UNIX_SECONDS(TIMESTAMP '2021-01-01 00:00:00')").returns(1609459200L).check();
+        assertQuery("select TYPEOF(?)").withParams(1L).returns("BIGINT").check();
+        assertQuery("select ? + ?").withParams(1, 2).returns(3).check();
+        assertThrows("select PLUS(?, ?)", SqlValidatorException.class, "No match found for function signature", 1, 2);
+
+        // Ensure that new functions are successfully created in a custom schema.
+        assertQuery("SELECT \"OWN_SCHEMA\".UPPER(?)").withParams("abc").returns(3).check();
+        assertQuery("select \"OWN_SCHEMA\".UNIX_SECONDS(TIMESTAMP '2021-01-01 00:00:00')").returns(1).check();
+        assertQuery("select \"OWN_SCHEMA\".TYPEOF('ABC')").returns(1).check();
+        assertQuery("select \"OWN_SCHEMA\".PLUS(?, ?)").withParams(1, 2).returns(100).check();
+
+        LogListener logChecker0 = LogListener.matches("Unable to add user-defined SQL function 'upper'")
+            .andMatches("Unable to add user-defined SQL function 'unix_seconds'")
+            .andMatches("Unable to add user-defined SQL function 'typeof'")
+            .andMatches("Unable to add user-defined SQL function 'plus'").times(0)
+            .build();
+
+        listeningLog.registerListener(logChecker0);
+
+        // Try to add the functions into the default schema.
+        client.getOrCreateCache(new CacheConfiguration<Integer, Employer>("TEST_CACHE_PUB")
+            .setSqlFunctionClasses(OverrideSystemFunctionLibrary.class)
+            .setSqlSchema(QueryUtils.DFLT_SCHEMA)
+            .setQueryEntities(F.asList(new QueryEntity(Integer.class, Employer.class).setTableName("emp_pub"))));
+
+        assertTrue(logChecker0.check(getTestTimeout()));
+
+        // Make sure that the standard functions work once again.
+        assertQuery("SELECT UPPER(?)").withParams("abc").returns("ABC").check();
+        assertQuery("select UNIX_SECONDS(TIMESTAMP '2021-01-01 00:00:00')").returns(1609459200L).check();
+        assertQuery("select TYPEOF(?)").withParams(1L).returns("BIGINT").check();
+
+        // Make sure that operator + works and new function 'PLUS' also regustered in the default schema.
+        assertQuery("select ? + ?").withParams(1, 2).returns(3).check();
+        assertQuery("select PLUS(?, ?)").withParams(1, 2).returns(100);
+
+        CalciteQueryProcessor qryProc = Commons.lookupComponent(client.context(), CalciteQueryProcessor.class);
+        Map<String, IgniteSchema> schemas = GridTestUtils.getFieldValue(qryProc, "schemaHolder", "igniteSchemas");
+        IgniteSchema dfltSchema = schemas.get(QueryUtils.DFLT_SCHEMA);
```

Comment:
Why not use public API?
queryProcessor(ignite).schemaHolder().schema(QueryUtils.DFLT_SCHEMA)